File: src/test/java/cz/jskrabal/proxy/support/ProxyTestUtils.java
package cz.jskrabal.proxy.support;

import io.vertx.core.http.HttpMethod;

import java.util.EnumSet;

/**
 * Created by janskrabal on 06/07/16.
 */
public class ProxyTestUtils {
	private ProxyTestUtils() {}

	public static final int HTTP_SERVER_PORT = 7070;
	public static final int PROXY_PORT = 8080;
	public static final String PROXY_HOST = "localhost";
	public static final String BASE_REQUEST_URI = "http://" + PROXY_HOST + ':' + HTTP_SERVER_PORT + '/';

	//Test HTTP server constants
	public static final String URL_PARAM_RESPONSE = "response";
	public static final String URL_PARAM_RESPONSE_CODE = "responseCode";
	public static final String URL_PARAM_RESPONSE_TYPE = "responseType";

	public static boolean canHaveBody(HttpMethod method) {
		EnumSet methodsThatRestrictsRequestBody = EnumSet.of(
				HttpMethod.OPTIONS,
				HttpMethod.CONNECT,
				HttpMethod.GET,
				HttpMethod.HEAD,
				HttpMethod.TRACE);

		return !methodsThatRestrictsRequestBody.contains(method);
	}
}

File: src/test/kotlin/cz/jskrabal/proxy/support/ProxyTestUtils.kt
package cz.jskrabal.proxy.support

import java.util.EnumSet

import io.vertx.core.http.HttpMethod

/**
 * Created by janskrabal on 06/07/16.
 */
object ProxyTestUtils {

    val HTTP_SERVER_PORT = 7070
    val PROXY_PORT = 7000
    val PROXY_HOST = "localhost"
    val BASE_REQUEST_URI = "http://$PROXY_HOST:$HTTP_SERVER_PORT/"

    //Test HTTP server constants
    val URL_PARAM_RESPONSE = "response"
    val URL_PARAM_RESPONSE_CODE = "responseCode"
    val URL_PARAM_RESPONSE_TYPE = "responseType"

    fun canHaveBody(method: HttpMethod): Boolean {
        val methodsThatRestrictsRequestBody = EnumSet.of(
                HttpMethod.OPTIONS,
                HttpMethod.CONNECT,
                HttpMethod.GET,
                HttpMethod.HEAD,
                HttpMethod.TRACE)

        return !methodsThatRestrictsRequestBody.contains(method)
    }
}
